lib/bundler/definition.rb

```diff
@@ -977,7 +977,9 @@ def additional_base_requirements_for_resolve
       dependencies_by_name = dependencies.inject({}) {|memo, dep| memo.update(dep.name => dep) }
       @locked_gems.specs.reduce({}) do |requirements, locked_spec|
         name = locked_spec.name
-        next requirements if @locked_gems.dependencies[name] != dependencies_by_name[name]
+        dependency = dependencies_by_name[name]
+        next requirements if @locked_gems.dependencies[name] != dependency
+        next requirements if dependency && dependency.source.is_a?(Source::Path)
```

Comment:
I think we still need to make this an `instance_of?` rather than a `is_a?` to avoid affecting `Source::Gemspec` and `Source::Git` dependencies.

Reply:
Why though? Those subclasses also dont have any way to resolve versions

Reply:
You're totally right - in fact we *want* downgrades of gemspec dependencies and changing this would break them. Have added a test (broken on master, fixed on this branch)

Reply:
Thanks for adding a test!